src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Primitives/WriterDelegate.cs

```diff
@@ -72,10 +72,11 @@ protected override void Dispose(bool disposing)
                 if (disposing && !IsDisposed)
                 {
                     _addDelegate(XamlNodeType.None, XamlNode.InternalNodeType.EndOfStream);
-                    _addDelegate = new XamlNodeAddDelegate(ThrowBecauseWriterIsClosed);
-                    _addLineInfoDelegate = (_addLineInfoDelegate != null)
-                                ? new XamlLineInfoAddDelegate(ThrowBecauseWriterIsClosed2)
-                                : null;
+                    _addDelegate = delegate { throw new XamlException(SR.Get(SRID.WriterIsClosed)); };
```

Comment:
Do we even need to set `_addDelegate` and `_addLineInfoDelegate` ? ThrowIsDisposed should already be called before the use of `_addDelegate` or `_addLineInfoDelegate`.

Reply:
I still think this PR is valid because it's faster than the existing code, which was already executed even if the new delegate is not invoked. A follow-up PR could be open to remove this "dead code" with more thorough testing to make sure that the delegates are never executed after dispose. I only said this over a quick glance over the code but they might be invoked.

Reply:
> I still think this PR is valid

It is :smile: